substrate/frame/system/src/extensions/weights.rs

```diff
@@ -68,74 +68,70 @@ impl<T: crate::Config> WeightInfo for SubstrateWeight<T> {
 	/// Proof: `System::BlockHash` (`max_values`: None, `max_size`: Some(44), added: 2519, mode: `MaxEncodedLen`)
 	fn check_genesis() -> Weight {
 		// Proof Size summary in bytes:
-		//  Measured:  `54`
+		//  Measured:  `30`
 		//  Estimated: `3509`
-		// Minimum execution time: 3_876_000 picoseconds.
-		Weight::from_parts(4_160_000, 3509)
+		// Minimum execution time: 3_388_000 picoseconds.
+		Weight::from_parts(3_577_000, 3509)
```

Comment:
How did you manage to not the count the `BlockHash` storage read?

I thought we would write something like:
```diff
diff --git a/substrate/frame/system/benchmarking/src/extensions.rs b/substrate/frame/system/benchmarking/src/extensions.rs
index 2c7ffb56227..d9564feb6d9 100644
--- a/substrate/frame/system/benchmarking/src/extensions.rs
+++ b/substrate/frame/system/benchmarking/src/extensions.rs
@@ -53,6 +53,9 @@ mod benchmarks {
                let caller = whitelisted_caller();
                let info = DispatchInfo { call_weight: Weight::zero(), ..Default::default() };
                let call: T::RuntimeCall = frame_system::Call::remark { remark: vec![] }.into();
+               frame_benchmarking::benchmarking::add_to_whitelist(
+                       frame_system::BlockHash::<T>::hashed_key_for(BlockNumberFor::<T>::from(0u32)).into(),
+               );
 
                #[block]
                {
```

Reply:
I did it with `whitelisted_caller` since it was the only read the bench did, but this is more granular so I replaced it with your suggestion.

Reply:
ok both implementation are good to me.

Reply:
why my suggestion wrong? there is still a read for the block hash here.